adr/00004-use-stimulus-as-our-javascript-framework.md

```diff
@@ -0,0 +1,23 @@
+# 4. Use Stimulus as our Javascript framework
+
+Date: 2024-01-23
+
+## Status
+
+Accepted
+
+## Context
+
+We want to better organise our JavaScript. Stimulus is the default js framework for rails and offers a lot more control then not having a js framework. It allows us to choose which pages or html elements do need a bit JavaScript.
+
+## Decision
+
+Use Stimulus controllers to enhance html elements or pages. This can be done via the stimulus-rails gem.
+- https://stimulus.hotwired.dev/
+- https://github.com/hotwired/stimulus-rails
+
+## Consequences
+
+As with any JavaScript framework or any code, there is a possibility of adding a lot of code and things can get hard to maintain. Luckily our app doesn't require too much JavaScript.
+
+If we broadly follow the rails conventions on name spacing/folder structure, we should be able not have many issues with this, a Stimulus controller can be reused in any view or just one view.
```

Comment:
```suggestion
If we broadly follow the Rails conventions on name spacing/folder structure, we should be able not have many issues with this, a Stimulus controller can be reused in any view or just one view.
```

Reply:
I'm not sure if it's best to put the general guidelines for Stimulus controllers here or not, but we should highlight that we want to create behaviour-driven controllers, not context-driven ones.

For example: `autocomplete-controller` not `schools-select-controller`. The former can then be configured to be used in multiple pages.

Reply:
I've moved it in the Decision